### ISSUE TITLE
p-columnFilter Accessibility Issues

### ISSUE THREAD

**Issue by srujana10499** [primefaces/primeng#18767]
### Describe the bug

The p-toggleSwitch component with checkboxes has accessibility violations when running the IBM Equal Accessibility Checker. The two issues are: The tabbable element's role 'columnheader' is not a widget role and UI component must have aria attributes.

### Reason for not contributing a PR

- [x] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/stackblitz-starters-fugentnu

### Environment

Stackblitz

### Angular version

20.0.4

### PrimeNG version

v20

### Steps to reproduce the behavior

1. Install the [IBM Equal Accessibility Checker](https://chromewebstore.google.com/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp)
2. Open the stackblitz
3. Run the accessibility checker
4. Filter on violations only and note the errors: 'The tabbable element's role 'columnheader' is not a widget role' and 'The ARIA attributes "aria-expanded" are not valid for the element <p-button> with ARIA role "none"'

### Expected behavior

There are no accessibility violations for the p-columnFilter component
